Implementation/Unit-Testing/index.md

```diff
@@ -4,39 +4,104 @@ versionFrom: 8.0.0
 
 # Unit Testing Umbraco
 
-These examples requires [NUnit](https://nunit.org/) and [Moq](https://github.com/moq/moq4).
+These examples are for Umbraco 8+ and they rely on [NUnit](https://nunit.org/) and [Moq](https://github.com/moq/moq4).
 
-## SetUp
+# Mocking
 
-The ```Current.Factory``` needs to be mocked before each unit test that has an Umbraco dependency, or you'll get an ```InvalidOperationException``` saying that  **"No factory has been set"**.
+When testing components in Umbraco, especially controllers, there are a few dependencies that needs to be mocked / faked in order to get your unit tests running. Every Umbraco controller has two constructors: one empty constructor without any parameters for anyone not interested in unit testing or dependency injections, and one with full constructor injection which contains all parameters needed for proper unit testing. A lot of these dependencies are interfaces, which are simply mocked using Mock.Of<>, but there are still a few explicit non-interface dependencies that needs to be faked. In this documentation all mocks and fakes have been placed in a base class to avoid having to repeat this setup in every test class.
 
 ```csharp
-[SetUp]
-public void SetUp()
+public abstract class UmbracoBaseTest 
 {
-    Current.Factory = new Mock<IFactory>().Object;
+    public ServiceContext ServiceContext;
+    public MembershipHelper MembershipHelper;
+    public UmbracoHelper UmbracoHelper;
+
+    public Mock<ICultureDictionary> CultureDictionary;
+    public Mock<ICultureDictionaryFactory> CultureDictionaryFactory;
+    public Mock<IPublishedContent> PublishedContent;
+    public Mock<IPublishedContentQuery> PublichedContentQuery;
+
+    [SetUp]
+    public virtual void SetUp()
+    {
+        this.SetupCultureDictionaries();
+        this.SetupPublishedContentQuerying();
+
+        this.ServiceContext = ServiceContext.CreatePartial();
```

Comment:
Might be worth adding notes to the CreatePartial call here so that people understand that you can inject any mocked services that you want there for testing.

Reply:
Added explination in latest commit.